tests/test_catalog_delete.py

```diff
@@ -0,0 +1,112 @@
+        self.assertEqual(mcmu.cm_namespace, "3")
+        self.assertEqual(mcmu.product_name, "4")
+        self.assertEqual(mcmu.product_version, "5")
+        self.assertEqual(mcmu.max_reties_for_main_cm, "6")
+        self.assertEqual(mcmu.max_reties_for_prod_cm, "6.1")
+        self.assertEqual(mcmu.key, "7")
+        self.assertEqual(mcmu.main_cm_fields, ["8", "9", "10"])
+        self.assertEqual(mcmu.product_cm_fields, ["11", "12", "13"])
+
+        del mcmu
+
+    def test_delete_from_both_config_map(self):
+        """Test cases to assert delete calls into both main and product cm"""
+        self.modify_config_map_util.key = None
+        self.modify_config_map_util.modify_config_map()
+
+        self.mock_modify_config_map.assert_has_calls(
+            calls=[
+                # main cm call
+                call(self.modify_config_map_util.main_cm,
+                     self.modify_config_map_util.cm_namespace,
+                     self.modify_config_map_util.product_name,
+                     self.modify_config_map_util.product_version,
+                     self.modify_config_map_util.key,
+                     self.modify_config_map_util.max_reties_for_main_cm,
+                     ),
+                # product cm call
+                call(self.modify_config_map_util.product_cm,
+                     self.modify_config_map_util.cm_namespace,
+                     self.modify_config_map_util.product_name,
+                     self.modify_config_map_util.product_version,
+                     self.modify_config_map_util.key,
+                     self.modify_config_map_util.max_reties_for_prod_cm,
+                     )]
+        )
+
+    def test_delete_from_main_config_map(self):
+        """Test cases to assert delete calls into both main cm"""
+        self.modify_config_map_util.key = "main_a"
+        self.modify_config_map_util.modify_config_map()
+
+        self.mock_modify_config_map.assert_called_once_with(
+            self.modify_config_map_util.main_cm,
+            self.modify_config_map_util.cm_namespace,
+            self.modify_config_map_util.product_name,
+            self.modify_config_map_util.product_version,
+            self.modify_config_map_util.key,
+            self.modify_config_map_util.max_reties_for_main_cm,
+        )
+
+    def test_delete_from_product_config_map(self):
+        """Test cases to assert delete calls into both product cm"""
+        self.modify_config_map_util.key = "prod_3"
+        self.modify_config_map_util.modify_config_map()
+
+        self.mock_modify_config_map.assert_called_once_with(
+            self.modify_config_map_util.product_cm,
+            self.modify_config_map_util.cm_namespace,
+            self.modify_config_map_util.product_name,
+            self.modify_config_map_util.product_version,
+            self.modify_config_map_util.key,
+            self.modify_config_map_util.max_reties_for_prod_cm,
+        )
+
+    def test_invalid_key(self):
```

Comment:
Add a test which passes in a non-string key (since those are also not valid)

Reply:
Modified as stated,